pkg/controller/directvolumemigration/validation.go

```diff
@@ -0,0 +1,245 @@
+		})
+		return nil
+	}
+
+	// Not ready
+	if !cluster.Status.IsReady() {
+		direct.Status.SetCondition(migapi.Condition{
+			Type:     SourceClusterNotReady,
+			Status:   True,
+			Category: Critical,
+			Message:  SourceClusterNotReadyMessage,
+		})
+	}
+	return nil
+}
+
+func (r ReconcileDirectVolumeMigration) validateDestCluster(direct *migapi.DirectVolumeMigration) error {
+	ref := direct.Spec.DestMigClusterRef
+
+	if !migref.RefSet(ref) {
+		direct.Status.SetCondition(migapi.Condition{
+			Type:     InvalidDestinationClusterRef,
+			Status:   True,
+			Reason:   NotSet,
+			Category: Critical,
+			Message:  InvalidDestinationClusterReferenceMessage,
+		})
+		return nil
+	}
+
+	// Check if clusters are unique
+	if reflect.DeepEqual(ref, direct.Spec.SrcMigClusterRef) {
+		direct.Status.SetCondition(migapi.Condition{
+			Type:     InvalidDestinationCluster,
+			Status:   True,
+			Reason:   NotDistinct,
+			Category: Critical,
+			Message:  InvalidDestinationClusterMessage,
+		})
+		return nil
+	}
+
+	cluster, err := migapi.GetCluster(r, ref)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	// Not found
+	if cluster == nil {
+		direct.Status.SetCondition(migapi.Condition{
+			Type:     InvalidDestinationClusterRef,
+			Status:   True,
+			Reason:   NotFound,
+			Category: Critical,
+			Message:  InvalidDestinationClusterReferenceMessage,
+		})
+		return nil
+	}
+
+	// Not ready
+	if !cluster.Status.IsReady() {
+		direct.Status.SetCondition(migapi.Condition{
+			Type:     DestinationClusterNotReady,
+			Status:   True,
+			Category: Critical,
+			Message:  DestinationClusterNotReadyMessage,
+		})
+	}
+	return nil
+}
+
+// TODO: Validate that storage class mappings have valid storage class selections
+// Leaving as TODO because this is technically already validated from the
+// migplan, so not necessary from directvolumemigration controller to be fair
+func (r ReconcileDirectVolumeMigration) validateStorageClassMappings(direct *migapi.DirectVolumeMigration) error {
+	return nil
+}
+
+func (r ReconcileDirectVolumeMigration) validatePVCs(direct *migapi.DirectVolumeMigration) error {
+	allPVCs := direct.Spec.PersistentVolumeClaims
+
+	// Check if PVCs were set
+	if allPVCs == nil {
+		direct.Status.SetCondition(migapi.Condition{
+			Type:     InvalidPVCs,
+			Status:   True,
+			Reason:   NotSet,
+			Category: Critical,
+			Message:  InvalidPVCsMessage,
+		})
+		return nil
+	}
+	// Get source cluster client
+	cluster, err := direct.GetSourceCluster(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	// Check if these PVCs actually exist on the source
+	// cluster
+	notFound := make([]string, 0)
+	for _, specPVC := range allPVCs {
+		// Check if pvc actually exists and is bound on source cluster
+		// TODO: Check if PVC is actually attached. We should
```

Comment:
we need to decide what is a good assumption for 1.4.0? 

Willl we tolerate PVCs that have running pods for data copy through this controller or will we assume that they should always be unattached to the pods(i.e. application pods should be quiesced)

Reply:
we have broken this out in a separate issue. Resolving this for now